x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/type/EsField.java

```diff
@@ -80,16 +80,9 @@ public EsField getExactField() {
         return this;
     }
 
-    /**
-     * Returns the precision of the field
-     * <p>
-     * Precision is the specified column size. For numeric data, this is the maximum precision. For character
-     * data, this is the length in characters. For datetime datatypes, this is the length in characters of the
-     * String representation (assuming the maximum allowed defaultPrecision of the fractional seconds component).
-     */
-    public int getPrecision() {
-        return esDataType.defaultPrecision;
-    }
+    //    public int getPrecision() {
```

Comment:
Why is this method now available only for Keywords?

Reply:
Because it's used only there. Having it all over does not provide any value.
Same as `DateEsField` which initially was suppose to maintain the date format however that's not the case so essentially it's just another `EsField`.

These APIs can be cleaned as not all properties are needed (including precision) and can be gathered into the module that needs them.